tests/helper_networks.rs

```diff
@@ -140,3 +213,13 @@ fn http_network() {
 fn https_network() {
     test_network(&[4430, 4431, 4432], true);
 }
+
+#[test]
+fn http_semi_honest_ipa() {
+    test_ipa(&[3003, 3004, 3005], IpaSecurityModel::SemiHonest, false);
+}
+
+#[test]
+fn https_semi_honest_ipa() {
+    test_ipa(&[4433, 4434, 4435], IpaSecurityModel::SemiHonest, true);
```

Comment:
this is a bit annoying - I know that there is no guaranteed way to pick an available port (at least on MacOS, on Linux it works pretty reliable with SO_REUSEPORT and rust appears to be using [it](https://github.com/rust-lang-nursery/rust-cookbook/issues/500)), but even best-effort approach with TcpListener should work here. Right now it is not possible to test this in parallel

Reply:
I thought about this when I wrote the test... a possible strategy I came up with is to create `TcpListeners` in the test, pass them to the helper processes as /dev/stdin, and add an option to the helper that does `TcpListener::from_raw_fd` rather than bind a socket. I didn't do this at the time because it seemed like a fair amount of work, and it seemed useful to have at least one test that exercises the standard flow for binding the socket on startup. But if we're adding more tests that follow this framework, it becomes more important to do this work. I'll see if I can get this done next week.